ddl/ddl_api.go

```diff
@@ -2528,10 +2529,17 @@ func (d *ddl) DropColumn(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTa
 }
 
 // modifiableCharsetAndCollation returns error when the charset or collation is not modifiable.
-func modifiableCharsetAndCollation(toCharset, toCollate, origCharset, origCollate string) error {
+// needRewriteCollationData is used when trying to modify the collation of a column, it is true when the column is with
+// index because index of a string column is collation-aware.
+func modifiableCharsetAndCollation(toCharset, toCollate, origCharset, origCollate string, needRewriteCollationData bool) error {
```

Comment:
Since this function return `error` instead of `bool`, we'd better rename it to `checkModifyCharsetAndCollation`

Reply:
Sure.